extensions/azurecore/src/azureResource/utils.ts

```diff
@@ -151,51 +155,74 @@ export async function runResourceQuery<T extends azureResource.AzureGraphResourc
 		return result;
 	}
 
-	if (!subscription.tenant) {
-		const error = new Error(localize('azure.accounts.runResourceQuery.errors.noTenantSpecifiedForSubscription', "Invalid tenant for subscription"));
-		if (!ignoreErrors) {
-			throw error;
+	// Check our subscriptions to ensure we have valid ones
+	subscriptions.forEach(subscription => {
+		if (!subscription.tenant) {
+			const error = new Error(localize('azure.accounts.runResourceQuery.errors.noTenantSpecifiedForSubscription', "Invalid tenant for subscription"));
+			if (!ignoreErrors) {
+				throw error;
+			}
+			result.errors.push(error);
 		}
-		result.errors.push(error);
+	});
+	if (result.errors.length > 0) {
 		return result;
 	}
 
-	const tokenResponse = await azdata.accounts.getAccountSecurityToken(account, subscription.tenant, azdata.AzureResource.ResourceManagement);
-	const token = tokenResponse.token;
-	const tokenType = tokenResponse.tokenType;
-	const credential = new TokenCredentials(token, tokenType);
+	// We need to get a different security token for each tenant to query the resources for the subscriptions on
+	// that tenant
+	for (let i = 0; i < account.properties.tenants.length; ++i) {
+		const tenant = account.properties.tenants[i];
+		const tenantSubscriptions = subscriptions.filter(subscription => subscription.tenant === tenant.id);
```

Comment:
@aaomidi Make sure my understanding of this is correct here - I can't imagine a valid scenario where a subscription passed in has a tenant that isn't on the account unless the subscription isn't actually associated with the account passed in (which is invalid anyways). 

Assuming this is correct then it makes sense to get the security token for each tenant and run the query for all the subscriptions for that tenant right?

Reply:
Yep that's perfectly fine.